### PR TITLE
Fix passing of lease settings in typed cluster sharding.

### PR DESCRIPTION
**Pull request by horvath-martin**
References #29662

### CODE REVIEW

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala [hunk 1]
```diff
@@ -353,7 +356,8 @@ final class ClusterShardingSettings(
       shardRegionQueryTimeout: FiniteDuration,
       stateStoreMode: ClusterShardingSettings.StateStoreMode,
       tuningParameters: ClusterShardingSettings.TuningParameters,
-      coordinatorSingletonSettings: ClusterSingletonManagerSettings) =
+      coordinatorSingletonSettings: ClusterSingletonManagerSettings,
+      leaseSettings: Option[LeaseUsageSettings]) =
```
Comment (johanandren) [inline akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala:360]: I think we will have to add another constructor rather, to keep source/binary compatiblity, and deprecate it (that will cover the current constructor, which user code may be calling)

**Review by johanandren — COMMENTED** [review]
One thing around binary compatibility. Make sure to compile before commit so you get the code autoformatted.

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala (hunk 1 above)
Comment (horvath-martin) [reply akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala:360]: I have compiled it, but the formatter changed 66 other files.
Should I also add these files as well?
format

Comment (johanandren) [reply akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala:360]: Hmmm, that is surprising, it should only change the files you have touched (and have failed the PR validation if anyone tried to commit incorrectly formatted sources just like how it did with this PR).

Skip those and only push the files you have changed, after they were reformatted.

**Review by johanandren — APPROVED** [review]
LGTM, thanks @horvath-martin !

File: akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala [hunk 2]
```diff
@@ -339,8 +341,37 @@ final class ClusterShardingSettings(
     val stateStoreMode: ClusterShardingSettings.StateStoreMode,
     val rememberEntitiesStoreMode: ClusterShardingSettings.RememberEntitiesStoreMode,
     val tuningParameters: ClusterShardingSettings.TuningParameters,
-    val coordinatorSingletonSettings: ClusterSingletonManagerSettings) {
+    val coordinatorSingletonSettings: ClusterSingletonManagerSettings,
+    val leaseSettings: Option[LeaseUsageSettings]) {
 
+  @deprecated("Use constructor with leaseSettings", "2.6.9")
```
Comment (patriknw) [inline akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala:347]: should be "2.6.10"

**Comment by horvath-martin** [comment]
Is there any further problem? The CI failed, but it seems to be a networking error. Can I somehow trigger a repeated build or is it even necessary? (The only change since the previous successful build was a text value change.)

**Review by chbatey — APPROVED** [review]

**Pull request merged**